corehq/apps/commtrack/parsing.py

```diff
@@ -123,7 +123,10 @@ def ledger_json_to_stock_report_helper(form, report_type, ledger_json):
     # helper functions
     def get_date():
         timestamp = ledger_json.get('@date') or form.received_on
-        if not isinstance(timestamp, datetime.datetime):
+        if type(timestamp) is datetime.date:
```

Comment:
To avoid the use of `type` you could change the order:
```
if isintance(ts, datetime):
    return timestamp
elif isinstance(ts, date):
    return datetime.datetime.combine(timestamp, datetime.time())
else:
    raise InvalidDate()
```